src/ducc0/infra/threading.cc

```diff
@@ -537,14 +573,20 @@ void Distribution::thread_map(std::function<void(Scheduler &)> f)
   latch counter(nthreads_);
   std::exception_ptr ex;
   std::mutex ex_mut;
+  // we "copy" the currently active thread pool to all executing threads
+  // during the time of execution and pop it again at the end.
+  // Not sure if this is necessary/useful though.
+  auto cur_pool = pool_manager.active_pool();
```

Comment:
This should probably replace `pool = get_pool()` above. They ought to be the same.

Reply:
Yes, absolutely. I left the `get_pool()` in for the moment, since replacing it everywhere increases the diff and perhaps detracts from the essential changes.

Reply:
Ah sorry! Now I understand more clearly what you meant. Should be fixed.